openedx/features/journals/api.py

```diff
@@ -68,18 +71,55 @@ class JournalsApiClient(object):
     """
     Class for interacting with the Journals Service
     """
-    def __init__(self, user):
+    def __init__(self):
         """
         Initialize an authenticated Enterprise service API client by using the
         provided user.
         """
-        self.user = user
-        jwt = JwtBuilder(user).build_token([])
+        self.user = self.get_journals_worker()
+        jwt = JwtBuilder(self.user).build_token(['email', 'profile'], 16000)
         self.client = EdxRestApiClient(
             configuration_helpers.get_value('JOURNALS_API_URL', settings.JOURNALS_API_URL),
             jwt=jwt
         )
 
+    def get_journals_worker(self):
+        """ Return journals worker """
+        return User.objects.get(username=JOURNAL_WORKER_USERNAME)
+
+
+def fetch_journal_access(site, user):
+    """
+    Retrieve journal access record for given user.
+    Retrieve if from the cache if present, otherwise send GET request to the journal access api
+        and store it in the cache
+
+    Args:
+        site (Site)
+        user (username | str): user to retrieve access records for
+
+    Returns:
+        list of dicts: list of journal access dicts
+
+    Raises:
+        ConnectionError: raised if lms is unable to connect to the journals service.
+        SlumberBaseException: raised if API response contains http error status like 4xx, 5xx etc...
+        Timeout: raised if API is talking to long to respond
+    """
+    api_resourse = 'journal_access'
+    cache_key = get_cache_key(
+        site_domain=site.domain,
+        resourse=api_resourse,
+        user=user
+    )
+
+    journal_access_records = cache.get(cache_key)
+    if not journal_access_records:
+        journal_access_records = JournalsApiClient().client.journalaccess.get(user=user)
+        cache.set(cache_key, journal_access_records, JOURNALS_CACHE_TIMEOUT)
```

Comment:
By caching the list of access records, will we run into an issue where a user will check the dashboard, purchase a journal, then end up on the dashboard after purchase and they journals won't show up for `JOURNALS_CACHE_TIMEOUT` seconds?

Reply:
right, good catch. how does LMS deal with this for course enrollments?

Reply:
we could have an JournalAccess endpoint in LMS as well as in Journals that gets called upon fulfillment. when called in LMS it could update the cache with the new record.
I'd say for now remove the caching and add a story on the backlog to deal with this

Reply:
@tuchfarber Good catch - i'll take the cache out for now and make a story for this